test/e2e-helm/e2e_helm_project_on_cluster_test.go

```diff
@@ -0,0 +1,282 @@
+				logOutput, err := tc.Kubectl.Logs(controllerPodName, "-c", "manager")
+				Expect(err).NotTo(HaveOccurred())
+				return logOutput
+			}
+			Eventually(managerContainerLogsAfterUpdateCR, time.Minute, time.Second).Should(
+				ContainSubstring("Upgraded release"))
+
+			By("Checking Deployment replicas spec is equals 2")
+			verifyReleaseUpgrade := func() error {
+				replicas, err := tc.Kubectl.Get(
+					false,
+					"deployment", releaseName, "-o", "jsonpath={..spec.replicas}")
+				Expect(err).NotTo(HaveOccurred())
+				if replicas != "2" {
+					return fmt.Errorf("release(CR) deployment with %s replicas", replicas)
+				}
+				return nil
+			}
+			Eventually(verifyReleaseUpgrade, time.Minute, time.Second).Should(Succeed())
+
+			By("Granting permissions to access the metrics and read the token")
+			_, err = tc.Kubectl.Command(
+				"create",
+				"clusterrolebinding",
+				fmt.Sprintf("metrics-%s", tc.TestSuffix),
+				fmt.Sprintf("--clusterrole=e2e-%s-metrics-reader", tc.TestSuffix),
+				fmt.Sprintf("--serviceaccount=%s:default", tc.Kubectl.Namespace))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Getting the token")
+			b64Token, err := tc.Kubectl.Get(
+				true,
+				"secrets",
+				"-o=jsonpath={.items[0].data.token}")
+			Expect(err).NotTo(HaveOccurred())
+			token, err := base64.StdEncoding.DecodeString(strings.TrimSpace(b64Token))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(token).NotTo(HaveLen(0))
+
+			By("Creating a Pod running the curl image")
+			cmdOpts := []string{
+				"run",
+				"curl",
+				fmt.Sprintf("--image=%s", curlImage),
+				"--restart=OnFailure",
+				"--", "curl", "-v", "-k", "-H", fmt.Sprintf(`Authorization: Bearer %s`, token),
+				fmt.Sprintf("https://e2e-%v-controller-manager-metrics-service.e2e-%v-system.svc:8443/metrics",
+					tc.TestSuffix, tc.TestSuffix),
+			}
+			_, err = tc.Kubectl.CommandInNamespace(cmdOpts...)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Ensuring curl Pod status")
+			verifyCurlUp := func() error {
+				By("validating the curl pod status")
+				status, err := tc.Kubectl.Get(
+					true,
+					"pods", "curl", "-o", "jsonpath={.status.phase}")
+				Expect(err).NotTo(HaveOccurred())
+				if status != "Completed" && status != "Succeeded" {
+					return fmt.Errorf("curl pod in %s status", status)
+				}
+				return nil
+			}
+			Eventually(verifyCurlUp, 3*time.Minute, time.Second).Should(Succeed())
+
+			By("Checking metrics endpoint serving as expected")
+			getCurlLogs := func() string {
+				logOutput, err := tc.Kubectl.Logs("curl")
+				Expect(err).NotTo(HaveOccurred())
+				return logOutput
+			}
+			Eventually(getCurlLogs, time.Minute, time.Second).Should(ContainSubstring("< HTTP/2 200"))
+
+			By("Ensuring the operator metrics contains a `resource_created_at` metric for the CR")
+			metricExportedCR := fmt.Sprintf("resource_created_at_seconds{group=\"%s\","+
+				"kind=\"%s\","+
+				"name=\"%s-sample\","+
+				"namespace=\"default\","+
```

Comment:
Since we're just running `kubectl` won't the namespace be the one defined in the current context? In travis, it's probably `default`, but when running locally, it could be anything.

We may need to create a separate namespace for the CR as part of the test and always do CR-related things in that namespace (this is what we were doing in the shell-based tests). That way this test works without relying on the user to use the `default` namespace in their current context.

Reply:
The test utils have methods which allow us to define in the namespace and outside of the namespace. When we say outside of the namespace it will use the default defined kubeconfig one. 

I solved it in another way without losing the advantages of the funcs available to do the tests since these changes would be required just for this tiny part of this check. See that before the check we get the ns where the CR is applied to use it here.

Reply:
That works. It still isn't _totally_ ideal because it runs a test in a namespace that the caller may not want to pollute, but we can look at improving this in a follow-up.